### PR TITLE
#3u4b2eq #CU-3u4b2ad zombienet builder running simplest dali devnet

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
## Issue

- nix run .#devnet-dali runs zombienet, CI tests works well (well on infra level, logically they incorrect)

https://app.clickup.com/t/3u4b2ad

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-3u4b2ad nixified network builded with hrmp, genesis, ports](https://app.clickup.com/t/3u4b2ad)

**Comment by andyjsbell** [comment]
@dzmitry-lahoda could we get a description on this please?

**Comment by dzmitry-lahoda** [comment]
@andyjsbell added

**Comment by andyjsbell** [comment]
Does it need to stay in *draft* ?

File: scripts/zombienet/default.nix [hunk 1]
```diff
@@ -63,12 +63,25 @@ in with prelude; {
...
+  mkRelaychainNode = { rpc_port, ws_port, name }:
+    {
+      #name = name; # "rococo-local-alice";
+      rpc_port = rpc_port; # 30444;
+      validator = true;
+      #ws_port = ws_port; # 9944
+    } // optionalAttrs (rpc_port != null) { inherit rpc_port; }
+    // optionalAttrs (ws_port != null) { inherit ws_port; };
+
+  mpRelaychainNodes = { chain, rpc_port ? 30444, ws_port ? 9944, count ? 2 }:
```
Comment (andyjsbell) [inline scripts/zombienet/default.nix:75]: should this be `mkRelaychainNodes` ?

Comment (dzmitry-lahoda) [reply scripts/zombienet/default.nix:75]: ))) yes.

**Comment by dzmitry-lahoda** [comment]
it runs end to end from nix . will clean up PR in the morning (builder works, need to start replacing to it in all flows)
image

**Review by KaiserKarel — APPROVED** [review]

**Review by obsessed-cake — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
@mergifyio refresh

**Comment by dzmitry-lahoda** [comment]
@Mergifyio update

**Comment by dzmitry-lahoda** [comment]
@mergifyio refresh

**Comment by dzmitry-lahoda** [comment]
@Dom-Roth @andresmechali may you help with this

```
  53 passing (33m)
  3 pending
  1 failing

  1) [SHORT] Vesting Pallet Tests
       #1.4  Multiple vested transfer beneficiaries (#1.1, #1.2, #1.3) can claim simultaneously.:

      AssertionError: expected '6000000000000' to equal '7000000000000'      
      actual expected
      
      60000000000007000000000000
      
      at vestedScheduleClaimVerifier.verifyVestedScheduleClaim (/home/runner/work/composable/composable/code/integration-tests/runtime-tests/test/tests/vesting/ts/test/tests/vesting/testHelper.ts:193:41)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at /home/runner/work/composable/composable/code/integration-tests/runtime-tests/test/tests/vesting/ts/test/tests/vesting/vestingTests.ts:481:9
      at Context.<anonymous> (/home/runner/work/composable/composable/code/integration-tests/runtime-tests/test/tests/vesting/ts/test/tests/vesting/vestingTests.ts:463:5)
```

I can create CU task if that fix.

Tests are brittle. They rely on initial balance == to some specific value. They should not. They should use any balance >= X and later check not absolute values but decrease and increces.

**Comment by dzmitry-lahoda** [comment]
I can ignore the test if that is OK too.

**Comment by dzmitry-lahoda** [comment]
@mergify refresh

**Pull request merged**